ui_testing/testcases/basic_tests/test06_orders.py

```diff
@@ -2118,3 +2118,36 @@ def test033_Duplicate_main_order_and_cahange_materiel_type(self):
         self.assertEqual(suborder_data['Material Type'], material_type)
         self.assertEqual(suborder_data['Article Name'], article)
         self.assertEqual(suborder_data['Test Units'], test_unit)
+
+    def test034_Duplicate_sub_order_and_cahange_materiel_type(self):
+        """
+        duplicate sub-order of any order then change the materiel type
+
+        LIMS-6227
+        """
+        self.info('get random main order data')
+        orders, payload = self.orders_api.get_all_orders(limit=50)
+        main_order = random.choice(orders['orders'])
+        self.order_page.search(main_order['orderNo'])
+        self.order_page.get_child_table_data()
+        self.info("duplicate the sub order from suborder's options")
+        self.order_page.duplicate_sub_order_from_table_overview()
+        self.info('change material type')
+        material_type = self.order_page.set_material_type()
+        self.info('Make sure that article and test units are empty')
+        self.assertEqual(self.base_selenium.get_value(element='order:article'), None)
+        self.assertEqual(self.base_selenium.get_value(element='order:test_unit'), None)
+        self.info('select random article and test unit')
+        article = self.order_page.set_article()
+        test_unit = self.order_page.set_test_unit()
```

Comment:
This could produce a bug easily. Once u select a random material type, This material type could have not any article. 

Get a test unit data via API and use it

Reply:
done